src/core/api/coap_api.cpp

```diff
@@ -218,11 +218,20 @@ otError otCoapSendRequestWithParameters(otInstance *              aInstance,
                                         void *                    aContext,
                                         const otCoapTxParameters *aTxParameters)
 {
+    otError   error;
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    return instance.GetApplicationCoap().SendMessage(*static_cast<Coap::Message *>(aMessage),
-                                                     *static_cast<const Ip6::MessageInfo *>(aMessageInfo),
-                                                     Coap::TxParameters::From(aTxParameters), aHandler, aContext);
+    if (aTxParameters != NULL)
+    {
+        VerifyOrExit(Coap::TxParameters::Validate(*aTxParameters), error = OT_ERROR_INVALID_ARGS);
```

Comment:
What about making this a non-static method and validating after the static cast?

Reply:
That would result in checking the default parameters again and again. I prefer  avoid the unnecessary checking. Thoughts?

Reply:
You can avoid validating if `aTxParameters` is NULL.

Reply:
Oh, yes! I will update soon.